plugins/woocommerce/src/Caches/OrderCountCacheService.php

```diff
@@ -44,6 +60,18 @@ public function update_on_new_order( $order_id, $order ) {
 			return;
 		}
 
+		// If the order status was updated, we need to increment the order count cache for the
+		// initial status that was errantly decremented on order status change.
+		if ( isset( $this->initial_order_statuses[ $order_id ] ) ) {
```

Comment:
Because `woocommerce_new_order` can fire before or after `woocommerce_order_status_changed` and `woocommerce_order_status_changed` can fire multiple times, we want to make sure we didn't accidentally decrement a status count that had never been persisted.